cala-server/src/graphql/convert.rs

```diff
@@ -82,6 +93,114 @@ impl From<cala_ledger::journal::JournalValues> for Journal {
     }
 }
 
+impl From<cala_ledger::tx_template::TxTemplateValues> for TxTemplate {
+    fn from(value: cala_ledger::tx_template::TxTemplateValues) -> Self {
+        let tx_input = TxInput::from(value.tx_input);
+        let entries = value.entries.into_iter().map(EntryInput::from).collect();
+        let params = value
+            .params
+            .map(|params| params.into_iter().map(ParamDefinition::from).collect());
+        Self {
+            id: value.id.to_global_id(),
+            tx_template_id: UUID::from(value.id),
+            code: value.code,
+            tx_input,
+            entries,
+            params,
+            description: value.description,
+            metadata: value.metadata.map(JSON::from),
+        }
+    }
+}
+
+impl From<cala_ledger::tx_template::TxInput> for TxInput {
+    fn from(
+        cala_ledger::tx_template::TxInput {
+            effective,
+            journal_id,
+            correlation_id,
+            external_id,
+            description,
+            metadata,
+        }: cala_ledger::tx_template::TxInput,
+    ) -> Self {
+        Self {
+            effective: Expression::from(effective),
+            journal_id: Expression::from(journal_id),
+            correlation_id: correlation_id.map(Expression::from),
+            external_id: external_id.map(Expression::from),
+            description: description.map(Expression::from),
+            metadata: metadata.map(Expression::from),
+        }
+    }
+}
+
+impl From<cala_ledger::tx_template::EntryInput> for EntryInput {
+    fn from(
+        cala_ledger::tx_template::EntryInput {
+            entry_type,
+            account_id,
+            layer,
+            direction,
+            units,
+            currency,
+            description,
+        }: cala_ledger::tx_template::EntryInput,
+    ) -> Self {
+        Self {
+            entry_type: Expression::from(entry_type),
+            account_id: Expression::from(account_id),
+            layer: Expression::from(layer),
+            direction: Expression::from(direction),
+            units: Expression::from(units),
+            currency: Expression::from(currency),
+            description: description.map(Expression::from),
+        }
+    }
+}
+
+impl From<cala_ledger::tx_template::ParamDefinition> for ParamDefinition {
+    fn from(value: cala_ledger::tx_template::ParamDefinition) -> Self {
+        let default = value.default.map(Expression::from);
+        Self {
+            name: value.name,
+            r#type: ParamDataType::from(value.r#type),
+            default,
+            description: value.description,
+        }
+    }
+}
+
+impl From<cala_ledger::tx_template::ParamDataType> for ParamDataType {
```

Comment:
I think async-graphql has an option to generate this conversion when you declare the GQL run type